### DISCUSSION TITLE
Persist current session to a file

### DISCUSSION THREAD

**Discussion by LiamKearn** [saulpw/visidata#2385] [Q&A] [answered]
Hi All,

How can I persist my current session to a file? E.g I want to save my column typings etc.

**Not asking how to save visidata state to a csv** 😆

Thanks in advance `:)`

**Comment by ajkerrigan** [comment, marked as answer]
You should be able to use `g` `Ctrl+s` to save all sheets, and use the `.vds` file extension. That's a VisiData-specific format that should help you avoid losing those column type customizations.

From the [VisiData Internal Formats](https://www.visidata.org/docs/internal_formats/) docs:

> .vds is a new, simple, multisheet format based on JSONL that includes some sheet metadata (mostly column definitions). If you care only about the data and columns on the sheets, and not any provenance or exactitude or any of that stuff you get from going directly to the source, then you can save to .vds. It should load generally to an index of the sheets, with each retaining the data and layout it was saved with, but without any history.

**Comment by LiamKearn** [comment]
@ajkerrigan This is what I was after, thanks 😁
